client/app/scripts/superdesk-publish/publish.js

```diff
@@ -176,17 +176,18 @@
                 $scope.remove = function(subscriber) {
                     modal.confirm(gettext('Are you sure you want to delete subscriber?'))
                     .then(function() {
-                        api.subscribers.remove(subscriber)
-                        .then(function(result) {
-                            _.remove($scope.subscribers, subscriber);
-                        }, function(response) {
-                            if (angular.isDefined(response.data._message)) {
-                                notify.error(gettext('Error: ' + response.data._message));
-                            } else {
-                                notify.error(gettext('There is an error. Subscriber cannot be deleted.'));
-                            }
-                        }).then(fetchSubscribers);
-                    });
+                        api.subscribers.remove(subscriber);
```

Comment:
Missing `return`? We must return the `api`'s promise so that the next `.then()` in the chain only gets executed once the API call has completed.